In [3]:
import re
import pandas as pd
import os
import time
from tqdm.notebook import tqdm
import unidecode
import numpy as np

from preprocessing_functions import *

In [12]:
language = 'fr' #'fr'
indir = f'../../../../data/'
inputfp = os.path.join(indir, f'{language}wiki/')

In [13]:
#- loop over .csv files create dataframes
df = pd.DataFrame(columns = ['article_id', 'title', 'text'])
# iterate over directory for each file path, create a dataframe
for file in os.listdir(inputfp):
    fp = os.path.join(inputfp, file)
    df_temp = pd.read_csv(fp)
    df = pd.concat([df, df_temp])

In [4]:
def count_pars(article):
    """count number of paragraphs in an article"""
    count = 0 # initialise count at 0

    # split article into paragraphs (by using '\n' as end of paragraph)
    paragraphs = article.splitlines()
    for paragraph in paragraphs:
        # if paragraph empty skip
        if not paragraph:
            continue

        # check if paragraph > 2
        length = len(paragraph.split(' '))
        if length >= 2:
            count += 1
    
    return count

In [14]:
# loop over multiple articles
total_count = 0
for article in tqdm(df.text, total = len(df.text), desc = "Articles processed"):
    total_count += count_pars(article)

    
print(f"Total number of paragraphs where a collocation could occur is: {total_count}")

Articles processed:   0%|          | 0/274578 [00:00<?, ?it/s]

Total number of paragraphs where a collocation could occur is: 6005355


number of paragraphs for **english**:\
Total number of paragraphs where a collocation could occur is: **10,436,996**  
  
  
  
number of paragraphs for **french**:\
Total number of paragraphs where a collocation could occur is: **6,005,355**